src/libraries/System.Text.Json/Common/ReflectionExtensions.cs

```diff
@@ -324,5 +322,102 @@ public static bool IsKeyValuePair(this Type type, Type? keyValuePairType = null)
 
             return defaultValue;
         }
+
+        /// <summary>
+        /// Returns a list containing the class hierarchy for the given type,
+        /// starting from the current type up to the least derived base type.
+        /// </summary>
+        public static Type[] GetSortedClassHierarchy(this Type type)
```

Comment:
>Sorted [](http://example.com/codeflow?start=32&length=6)

nit: the word sorted here is a little bit confusing. Maybe something like `GetClassHierarchyChain` be clearer?

Reply:
The name reflects the fact that the types are topologically sorted. This is not particularly meaningful for classes which always have chained hierarchies like you mention, however the name simply reflects the `GetSortedInterfaceHierarchy` which might not be a chain and returns a topologically sorted sequence.

Reply:
Yes, I saw that after continue reviewing the code :-)